src/shell.js

```diff
@@ -272,11 +272,11 @@ if (typeof console !== 'undefined') {
   Module['printErr'] = console.warn;
 } else if (typeof print !== 'undefined') {
   Module['print'] = print;
+}
+if (!Module['printErr']) {
   if (typeof printErr !== 'undefined') Module['printErr'] = printErr;
-} else {
-  Module['print'] = {};
+  else Module['printErr'] = Module['print'];
```

Comment:
Perhaps even a single line

```javascript
Module['printErr'] = Module['printErr'] || (typeof printErr !== 'undefined' ? printErr : Module['print']);
```

which could look a bit cleaner even, and is 106 chars over 144 chars?